exercise_2/Graph.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Graph.h"
+#include <iostream>
+
+Graph::Graph(int(&a)[N][N])
+{
+	for (int i = 0; i < N; ++i)
+	{
+		for (int j = 0; j < N; ++j)
+		{
+			arr[i][j] = a[i][j];
+		}
+	}
+}
+
+Graph::~Graph()
+{
+}
+
+void  Graph::printGraph()
+{
+	for (int i = 0; i < N; ++i)
+	{
+		for (int j = 0; j < N; ++j)
+		{
+			std::cout << arr[i][j] << " ";
+		}
+		std::cout << std::endl;
+	}
+}
+
+void Graph::printPath(int a[N], int start)
+{
+	for (int i = 0; i < N; i++)
+	{
+		std::cout << "From " << start << " to " << i << "  " << a[i] << " " << std::endl;
+	}
+}
+
+void  Graph::findShortestPath(int start = 0)
+{
+	int path[N]{ 0 };
+	bool visited[N];
+
+	for (int i = 0; i < N; ++i)
+	{
+		path[i] = INT_MAX;
+		visited[i] = false;
+	}
+	path[start] = 0;
+	int index = -1;
+	for (int i = 0; i < N; ++i)
```

Comment:
This is nice cascade:
https://www.profit24.pl/images/editor/ZABAWKI/Castorland/5904438103782/5904438103782_1.jpg

Usualy, we do not do such cascade in code. Could you consider to extract some code to separate functions?

Reply:
ok, I understand you.